### PR TITLE
Fix MDPDataset Discrete Action Update

### PR DESCRIPTION
**Pull request by jamartinh**
Fixing MDPDataset append method to allow grow action size when actions are discrete

### CODE REVIEW

**Review by takuseno — CHANGES_REQUESTED** [review]
Thanks! Can you also a simple unit test for this?

**Comment by jamartinh** [comment]
Yes of course

File: tests/test_dataset.py [hunk 1]
```diff
@@ -146,17 +147,38 @@ def test_mdp_dataset(
...
     if discrete_action:
-        assert dataset.actions.shape == (2 * data_size,)
+        assert dataset.actions.shape == (new_size * data_size,)
     else:
-        assert dataset.actions.shape == (2 * data_size, action_size)
+        assert dataset.actions.shape == (new_size * data_size, action_size)
+
+    # check append if discrete action and number of actions grow
+    if discrete_action:
+        new_size += 1
```
Comment (takuseno) [inline tests/test_dataset.py:166]: Thanks for the test. But, I actually want you to test specifically if `dataset.get_action_size()` is updated based on the new data. You can simplify this part.

**Review by takuseno — CHANGES_REQUESTED** [review]
@jamartinh Sorry for the late response and thank you for the changes. I'd like you to simplify the test only to make sure if the action size is correctly updated without exceptions.

File: tests/test_dataset.py (hunk 1 above)
Comment (jamartinh) [reply tests/test_dataset.py:166]: Ok, I added the specific check for dataset.get_action_size() == old size + new actions

File: tests/test_dataset.py [hunk 2]
```diff
@@ -146,17 +153,43 @@ def test_mdp_dataset(
...
+        with pytest.warns(UserWarning):
+            dataset.append(
+                observations, actions + add_actions, rewards, terminals
+            )
+            message = f"New action size is higher than" f" {ref_action_size}."
+            warnings.warn(message, UserWarning)
+
+        assert dataset.get_action_size() == old_action_size + add_actions
+        assert len(dataset) == new_size * n_episodes
```
Comment (takuseno) [inline tests/test_dataset.py:182]: You can remove all assertions except `dataset.get_action_size() == old_action_size + add_actions`.

File: tests/test_dataset.py [hunk 3]
```diff
@@ -146,17 +153,43 @@ def test_mdp_dataset(
...
+        assert dataset.actions.shape == (new_size * data_size,)
     else:
-        assert dataset.actions.shape == (2 * data_size, action_size)
+        assert dataset.actions.shape == (new_size * data_size, action_size)
+
+    # check append if discrete action and number of actions grow
+    if discrete_action:
+        old_action_size = dataset.get_action_size()
+        new_size += 1
+        with pytest.warns(UserWarning):
```
Comment (takuseno) [inline tests/test_dataset.py:174]: You don't need to check warning.

**Review by takuseno — CHANGES_REQUESTED** [review]
Sorry for bothering you. But, I'd like you to check the comments.

File: tests/test_dataset.py (hunk 2 above)
Comment (takuseno) [reply tests/test_dataset.py:182]: Since we already tested in the previous `append`.

File: tests/test_dataset.py (hunk 3 above)
Comment (jamartinh) [reply tests/test_dataset.py:174]: ok, It was just to not having that warning on pytest

**Review by takuseno — APPROVED** [review]
LGTM. Thanks for the change!

**Pull request merged**